getting-started/src/main/kotlin/com/google/cloud/kotlin/demo/EmojifyController.kt

```diff
@@ -0,0 +1,166 @@
+    fun emojify(@RequestParam(value = "objectName") objectName: String): EmojifyResponse? {
+
+        var publicUrl: String =
+            "https://storage.googleapis.com/${bucket.name}/emojified/emojified-$objectName" // api response
+        val log = Logger.getLogger(Application::class.java.name)
+
+        // Setting up image annotation request
+        val requests = ArrayList<AnnotateImageRequest>()
+        val vision = ImageAnnotatorClient.create()
+        val source = ImageSource.newBuilder().setGcsImageUri("gs://${bucket.name}/$objectName").build() // Fetching our source image
+        val img = Image.newBuilder().setSource(source).build()
+        val feat = Feature.newBuilder().setType(Type.FACE_DETECTION).build()
+        val request = AnnotateImageRequest.newBuilder()
+                .addFeatures(feat)
+                .setImage(img)
+                .build()
+
+        requests.add(request)
+
+        // Calls vision api on above image annotation requests
+        val response = vision.batchAnnotateImages(requests)
+        vision.close()
+
+        for (resp in response.responsesList) {
+            if (resp.hasError()) {
+                log.severe(resp.error.message)
+                return null
+            }
+
+            downloadObject(objectName) // image is now downloaded to /tmp/<objectName>
+            val imgBuff = ImageIO.read(Paths.get("/tmp/$objectName").toFile())
+            val gfx = imgBuff.createGraphics()
+            val extension = (Paths.get("/tmp/$objectName").toFile().extension)
+
+            for (annotation in resp.faceAnnotationsList) {
+                val emoji = bestEmoji(annotation)
+                val imgEmoji = ImageIO.read(Paths.get(EMOJIS_PATH + emojiPic[emoji]).toFile())
+
+                log.info("""
+                    joy: ${annotation.joyLikelihood}
+                    anger: ${annotation.angerLikelihood}
+                    surprise: ${annotation.surpriseLikelihood}
+                    sorrow: ${annotation.sorrowLikelihood}
+                    position: ${annotation.boundingPoly}
+                """)
+
+                log.info("EMOJI DETECTED: $emoji")
+                val poly = Polygon()
+                for (vertex in annotation.fdBoundingPoly.verticesList) {
+                    poly.addPoint(vertex.x, vertex.y)
+                }
+
+                val height = poly.ypoints[2] - poly.ypoints[0]
+                val width = poly.xpoints[1] - poly.xpoints[0]
+                // Draws emoji on detected face
+                gfx.drawImage(imgEmoji, poly.xpoints[0], poly.ypoints[1], height, width, null)
+            }
+            // Writing emojified image to temp file
+            val writeTo = "/tmp/emojify-$objectName"
+            ImageIO.write(imgBuff, extension, Paths.get(writeTo).toFile())
+
+            // Uploading emojified image to GCS
+            val blob = bucket.create("emojified/emojified-$objectName", Files.readAllBytes(Paths.get(writeTo)))
```

Comment:
We could set the predefined ACL here instead of having to make a separate API call below. Trying to [decipher this](https://github.com/GoogleCloudPlatform/google-cloud-java/blob/master/google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java#L60):

```kt
import com.google.cloud.storage.Storage.PredefinedAcl;
import com.google.cloud.storage.Storage.BlobTargetOption

// ...
bucket.create(
    "emojified/emojified-$objectName",
    Files.readAllBytes(Paths.get(writeTo))
    BlobTargetOption.predefinedAcl(PredefinedAcl.PUBLIC_READ)
)
```